File: C#/CastleGame/Room.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CastleGame
{
    class Room
    {
        private string name;
        private string description;
        private Dictionary<string,string> neighbours;
        public Room(string name, string description, Dictionary<string,string> neighbours)
        {
            SetName(name);
            SetDescription(description);
            this.neighbours = new Dictionary<string, string>();
            setNeighbours(neighbours);
        }

        private void SetName (string name)
        {
            this.name = name;
        }

        public string GetName()
        {
            return this.name;
        }

        private void SetDescription(string description)
        {
            this.description = description;
        }

        public string GetDesciption()
        {
            return this.description;
        }

        private void setNeighbours(Dictionary<string,string> neighbours)
        {
            this.neighbours = neighbours;
        }

        public Dictionary<string,string> getNeighbours()
        {
            return this.neighbours;
        }
    }
}

File: Java/Visual/src/Room.java
import java.util.HashMap;

public class Room {


    private String name;
    private HashMap<String,String> neighbors;
    private String description;

    public Room(String name, HashMap<String,String> neighbors, String description) {
        setName(name);
        setNeighbors(neighbors);
        setDescription(description);
    }

    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }

    public HashMap<String, String> getNeighbors() {
        return neighbors;
    }
    public void setNeighbors(HashMap<String, String> neighbors) {
        this.neighbors = neighbors;
    }

    public String getDescription() {
        return description;
    }
    public void setDescription(String description) {
        this.description = description;
    }
}
